src/arrayMethodJoin.js

```diff
@@ -4,8 +4,24 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  Array.prototype.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    if (separator !== undefined) {
```

Comment:
now no need to do this check, you already add default value.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters
```suggestion
```